File: isSymmetric.swift
// Time Complexity : O(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach

    func isSymmetric(_ root: TreeNode?) -> Bool {
        guard let root = root else { return true }
        return isSymmetricHelper(root.left,root.right)
    }
    func isSymmetricHelper(_ n1: TreeNode?, _ n2: TreeNode?) -> Bool {
        if n1 == nil && n2 == nil {
            return true
        }
        if n1 == nil || n2 == nil || n1?.val != n2?.val {
            return false
        }
        return isSymmetricHelper(n1?.left,n2?.right) && isSymmetricHelper(n1?.right,n2?.left)
    }

File: isSymmetric.cpp
// Time Complexity : o(n)
// Space Complexity : o(n)
// Where n = number of elements in the list given
// Did this code successfully run on Leetcode : YES
// Any problem you faced while coding this : NO

class Solution {
public:
    bool helper(TreeNode* leftRoot, TreeNode* rightRoot) {
        if(leftRoot == NULL && rightRoot == NULL) {
            return true;
        }
        if(leftRoot!=NULL && rightRoot==NULL || leftRoot==NULL && rightRoot!=NULL) {
            return false;
        }
        if(leftRoot->val != rightRoot->val) {
            return false;
        }
        return helper(leftRoot->right, rightRoot->left) && helper(leftRoot->left, rightRoot->right);    
    }
    bool isSymmetric(TreeNode* root) {
        if(root==NULL) {
            return true;
        }
        return helper(root->left, root->right);
    }
};